src/Engine/Lod/Lod.h

```diff
@@ -0,0 +1,105 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "Version.h"
+
+
+namespace Lod {
+enum class OpenMode {
+    Read,
+    ReadWrite,
+};
+
+
+class Lod {
+public:
+    /**
+     * Open an existing LOD file.
+     *
+     * This function loads an existing LOD file for either reading or reading+writing.
+     *
+     * @param filename          LOD filename to open.
+     * @param mode              Opening mode.
+     */
+    static std::shared_ptr<Lod> open(const std::string &filename, OpenMode mode);
+
+    /**
+     * Creating a new LOD file.
+     *
+     * This function creates (potentially overwriting) a new LOD file.
+     *
+     * @param filename          LOD filename to create/overwrite.
+     * @param description       Brief LOD description (mandatory in the file header).
+     * @param version           LOD file version to create.
+     */
+    static std::shared_ptr<Lod> create(const std::string &filename, const std::string &description, Version version);
+
+    /**
+     * Closing the LOD file.
+     *
+     * This function closes the file and frees all associated resources, commiting any pending changes when in Write mode.
+     */
+    void close();
+
+    /**
+     * Changes the current directory.
+     *
+     * This function will change the current active directory in LOD.
+     *
+     * @param path              Path (can be relative to the current directory).
+     */
+    bool changeDirectory(const std::string& path);
```

Comment:
This implies that LOD is both the directory handle, and the directory iterator. This is usually not a great idea, a reader with no changeable state would be better.